api/reviews.go

```diff
@@ -134,3 +137,61 @@ func (api *API) getReviewHandler(writer http.ResponseWriter, request *http.Reque
 
 	log.Event(ctx, "successfully retrieved review", log.INFO, logData)
 }
+
+func (api *API) updateReviewHandler(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	bookID := mux.Vars(request)["id"]
+	reviewID := mux.Vars(request)["reviewID"]
+
+	logData := log.Data{"book_id": bookID, "review_id": reviewID}
+
+	if bookID == "" {
+		handleError(ctx, writer, apierrors.ErrEmptyBookID, logData)
+		return
+	}
+
+	if reviewID == "" {
+		handleError(ctx, writer, apierrors.ErrEmptyReviewID, logData)
+		return
+	}
+
+	// Confirm that book exists. If bookID not found, then do not check for the review
+	_, err := api.dataStore.GetBook(ctx, bookID)
+	if err != nil {
+		handleError(ctx, writer, err, logData)
+		return
+	}
+
+	// Confirm that the review exists
+	_, err = api.dataStore.GetReview(ctx, reviewID)
+	if err != nil {
+		handleError(ctx, writer, err, logData)
+		return
+	}
+
+	review := &models.Review{User: models.User{}}
+	if err := ReadJSONBody(ctx, request.Body, review); err != nil {
+		handleError(ctx, writer, apierrors.ErrInvalidReview, logData)
+		return
+	}
+
+	logData["review"] = review
+
+	err = api.dataStore.UpdateReview(ctx, reviewID, review)
+	if err != nil {
+		handleError(ctx, writer, err, logData)
+		return
+	}
+
+	review, err = api.dataStore.GetReview(ctx, reviewID)
+	if err != nil {
+		handleError(ctx, writer, err, logData)
+		return
+	}
```

Comment:
Is there a reason that you need to call get after updating? If there are updates to the review object during `UpdateReview` maybe consider updating the func to return the model as well as an err.

Reply:
Yeah, I guess I wanted to make sure the `UpdateReview` had _actually_ happened, and my way to check was to `GetReview` from the database

Reply:
I think it's enough to just check if an error has occurred to determine whether the update has taken place. I can see why you might want to return the updated representation in the response, but it looks like it's standard for a PUT to not return anything in the body of the response: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

If there was a reason to return an updated response in the body, then I agree with Dave's suggestion. Instead of calling `dataStore.GetReview` again, you could change the call to `dataStore.UpdateReview` to return the updated representation.

Reply:
Thanks both! I've double check and there is actually no reason for the model to be returned, so I've removed the second `get`